test/integration/cluster/cluster_test.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	informers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	clusterv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	clientset "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+	client "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
+)
+
+var clusterSpec = &clusterv1alpha1.ClusterSpec{
+	ClusterNetwork: clusterv1alpha1.ClusterNetworkingConfig{
+		ServiceDomain: "mydomain.com",
+		Services: clusterv1alpha1.NetworkRanges{
+			CIDRBlocks: []string{"10.96.0.0/12"},
+		},
+		Pods: clusterv1alpha1.NetworkRanges{
+			CIDRBlocks: []string{"192.168.0.0/16"},
+		},
+	},
+}
+
+// Constructs a function for async event retrieval
```

Comment:
godoc for functions requires that the first word of the comment match the name of the function. How about something like this?
```go
// checkEvents returns a function that attempts to retrieve an event from c. The returned function is compatible with Gomega's Eventually assertion.
```

Reply:
True. Thanks.